pkgs/development/tools/build-managers/gnumake/default.nix

```
@@ -32,6 +32,18 @@ stdenv.mkDerivation rec {
     ./0002-remove-impure-dirs.patch
   ];
 
+  # Remove unnecessary reference to C++ compiler. Note that MAKE_CXX is
+  # always set to $CXX when running confugure script. If it is not set,
+  # a reasonable fallback is used.
```

```suggestion
  # always set to $CXX when running Autotools-based configure script,
  # falling back to a reasonable fallback when CXX is not set.
```

P.S.: the "it" refers to what? CXX, as I have  wrote above?

---

If MAKE_CXX is not set (a preprocessor macro that is set to a build-time CXX value), the embedded default value for run-time CXX uses a reasonable fallback like `g++` (or `gpp` on MSDOS). I’ve updated the comment to elaborate on this behavior on more details.

---

I’ve updated the PR to use autoreconfHook with a patch to configure.ac.